tests/test_xarray.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+
+import pytest
+import xarray as xr
+
+import coincident
+from coincident.io.xarray import plot_esa_worldcover, to_dataset
+
+# These tests are based on the workflow from https://coincident.readthedocs.io/en/latest/examples/contextual_data.html
+# I couldn't think of a way to generalized these tests since the
+# xarray funcs are pretty tailored to COP30 and ESA WC
+
+# mypy doesn't like the resolution argument in to_dataset
+# Argument "resolution" to "to_dataset" has incompatible type "float"; expected "dict[str, Any]"  [arg-type]
```

Comment:
I'm admittedly pretty new to mypy, so there will be some things to figure out together :) In this case the problem is here:

https://github.com/uw-cryo/coincident/blob/c5a1a8521fe853420c0434b330159d1171ca7231/src/coincident/io/xarray.py#L39-L40

I expected that to work, but clearly it doesn't like the kwarg `resolution=0.008` in your test. 

Based on this https://stackoverflow.com/questions/37031928/type-annotations-for-args-and-kwargs, and quickly looking at the Xarray codebase, I think the easiest fix is to change `**kwargs: dict[str, Any]` -> `**kwargs: Any`, which will just disable typechecking for any keyword args passed to https://odc-stac.readthedocs.io/en/latest/_api/odc.stac.load.html#odc-stac-load